### PR TITLE
e2e: transfer receive enabled param test

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

```
make e2e-test entrypoint=TestTransferTestSuite test=TestReceiveEnabledParam
```

closes: #2014 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: e2e/testconfig/testconfig.go [hunk 1]
```diff
@@ -132,5 +138,38 @@ func newDefaultSimappConfig(cc ChainConfig, name, chainID, denom string) ibc.Cha
 		GasAdjustment:  1.3,
 		TrustingPeriod: "508h",
 		NoHostMount:    false,
+		ModifyGenesis:  modifyGenesisVotingPeriod("10s", denom),
+	}
+}
+
+func modifyGenesisVotingPeriod(votingPeriod string, denom string) func([]byte) ([]byte, error) {
```
Comment (colin-axner) [inline e2e/testconfig/testconfig.go:145]: I will split this out and clean it up once I figure out how to unmarshal the app state without using dyno

File: e2e/transfer_test.go [hunk 2]
```diff
@@ -208,6 +241,159 @@ func (s *TransferTestSuite) TestMsgTransfer_Timeout_Nonincentivized() {
...
+	t.Run("ensure transfer receive is enabled", func(t *testing.T) {
+		enabled := s.QueryTransferReceiveEnabledParam(ctx, chainA)
+		s.Require().True(enabled)
+	})
+
+	t.Run("ensure packets can be received, send from chainB to chainA", func(t *testing.T) {
+		t.Run("send from chainB to chainA", func(t *testing.T) {
+			transferTxResp, err := s.Transfer(ctx, chainB, chainBWallet, channelA.Counterparty.PortID, channelA.Counterparty.ChannelID, testvalues.DefaultTransferAmount(chainBDenom), chainBAddress, chainAAddress, s.GetTimeoutHeight(ctx, chainA), 0)
+			s.Require().NoError(err)
+			s.AssertValidTxResponse(transferTxResp)
```
Comment (colin-axner) [inline e2e/transfer_test.go:324]: This technically does not assert that the transaction succeeded. When it does, it'll probably be unnecessary to assert that the tokens were escrowed

**Review by damiannolan — APPROVED** [review]
Super clean! LGTM

**Review by chatton — APPROVED** [review]
LGTM great job 🥇

**Pull request merged**